### PR TITLE
GB-1500 - Login

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Features

- Implements CLI login

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: cli/crates/backend/src/login.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct TokenQueryParams {
+    token: String,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Credentials<'a> {
+    token: &'a str,
```
Comment (yoav-lavi) [inline cli/crates/backend/src/login.rs:32]: Still debating the contents / fields here

File: cli/crates/cli/src/login.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    thread::{sleep, spawn},
+    time::Duration,
+};
+
+pub fn login() -> Result<(), CliError> {
+    let (message_sender, message_receiver) = channel();
+
+    let login_handle = spawn(|| login::login(message_sender).map_err(CliError::BackendError));
+
+    if let Ok(LoginMessage::CallbackUrl(url)) = message_receiver.recv() {
```
Comment (yoav-lavi) [inline cli/crates/cli/src/login.rs:16]: `LoginMessage::CallbackUrl(url)` has to be the first message from `login`, debating structuring this differently

**Comment by yoav-lavi** [comment]
May change in accordance with @jvidalv developing the FE / minor updates / possibly the logout command (which deletes the credential file)

**Comment by yoav-lavi** [comment]
Testing will be tricky as it involves a real user and a web browser, needs to be considered

**Review by obmarg — APPROVED** [review]

**Pull request merged**